projects/fundamental-ngx/src/lib/calendar/calendar.component.ts

```diff
@@ -0,0 +1,485 @@
+                    this.selectedRangeLast = tempSelectedRangeFirst;
+                }
+            }
+        }
+        this.constructCalendar();
+    }
+
+    selectMonth(selectedMonth) {
+        this.selectedMonth = selectedMonth;
+        this.setCurrentMonth(selectedMonth);
+        this.constructCalendar();
+    }
+
+    setCurrentYear(year: number) {
+        this.date.setFullYear(year);
+        this.year = this.date.getFullYear();
+    }
+
+    selectYear(selectedYear) {
+        this.setCurrentYear(selectedYear);
+        this.selectedMonth = this.month;
+        this.constructCalendar();
+    }
+
+    //Functions that handle the calendar content - show/hide calendar dates, months list, years list
+    openMonthSelection() {
+        if (this.showCalendarYears) {
+            this.showCalendarYears = false;
+            this.showCalendarMonths = true;
+            this.showCalendarDates = false;
+        } else {
+            this.showCalendarMonths = !this.showCalendarMonths;
+            this.showCalendarYears = false;
+            this.showCalendarDates = !this.showCalendarDates;
+        }
+    }
+
+    openYearSelection() {
+        if (this.showCalendarMonths) {
+            this.showCalendarMonths = false;
+            this.showCalendarYears = true;
+            this.showCalendarDates = false;
+        } else {
+            this.showCalendarYears = !this.showCalendarYears;
+            this.showCalendarMonths = false;
+            this.showCalendarDates = !this.showCalendarDates;
+        }
+    }
+
+    @HostListener('document:keydown.escape', [])
+    onEscapeKeydownHandler() {
+        this.showCalendarDates = true;
+        this.showCalendarMonths = false;
+        this.showCalendarYears = false;
+    }
+
+    @HostListener('document:click', ['$event'])
+    onClickHandler(e: MouseEvent) {
+        const target = e.target;
+        if (!this.eRef.nativeElement.contains(target)) {
+            this.showCalendarDates = true;
+            this.showCalendarMonths = false;
+            this.showCalendarYears = false;
+        }
+    }
+
+    validateDateFromDatePicker(date: Array<number>): boolean {
+        let isInvalid: boolean = false;
+        let month = date[0];
+        let day = date[1];
+        let year = date[2];
+        let numOfDaysInMonth = 0;
+
+        if (isNaN(month) || isNaN(day) || isNaN(year)) {
+            isInvalid = true;
+            return isInvalid;
+        }
+
+        if (year < 1000 || year > 3000 || month < 1 || month > 12) {
+            isInvalid = true;
+            return isInvalid;
+        } else {
+            numOfDaysInMonth = this.daysPerMonth[month - 1];
+            if (day < 1 || day > numOfDaysInMonth) {
+                isInvalid = true;
+                return isInvalid;
+            }
+        }
+        return isInvalid;
+    }
+
+    ngOnChanges(changes: SimpleChanges) {
+        if(changes.dateFromDatePicker) {
```

Comment:
If I select a date, then edit it in the date picker input, the new date is selected on the calendar, but the blue outline still shows on the first selected date (see attachment.  might be intentional?).  However if I type in a date from scratch, the date is not visually selected in the calendar until I move the calendar's month to a different month and come back to the month for the date I typed in.  
![screen shot 2018-08-15 at 11 23 42 am](https://user-images.githubusercontent.com/2471874/44162503-cb7b1f00-a07d-11e8-861a-324e0cfabf5b.png)

Reply:
Yes, it's intentional. When you change the month and/or year you need to select the day to "trigger" the new date. It's up to interpretation/logic, so maybe we should discuss it.